airflow/ui/src/components/Table.tsx

```diff
@@ -0,0 +1,182 @@
+  */
+  setSortBy?: (sortBy: SortingRule<object>[]) => void;
+  pageSize?: number;
+}
+
+const Table: React.FC<Props> = ({
+  data, columns, manualPagination, pageSize = 25, setSortBy,
+}) => {
+  const { totalEntries, offset, setOffset } = manualPagination || {};
+  const oddColor = useColorModeValue('gray.50', 'gray.900');
+  const hoverColor = useColorModeValue('gray.100', 'gray.700');
+
+  const pageCount = totalEntries ? (Math.ceil(totalEntries / pageSize) || 1) : data.length;
+
+  const {
+    getTableProps,
+    getTableBodyProps,
+    allColumns,
+    prepareRow,
+    page,
+    canPreviousPage,
+    canNextPage,
+    nextPage,
+    previousPage,
+    state: { pageIndex, sortBy },
+  } = useTable(
+    {
+      columns,
+      data,
+      pageCount,
+      manualPagination: !!manualPagination,
+      manualSortBy: !!setSortBy,
+      initialState: {
+        pageIndex: offset ? offset / pageSize : 0,
+        pageSize,
+      },
+    },
+    useSortBy,
+    usePagination,
+  );
+
+  const handleNext = () => {
+    nextPage();
+    if (setOffset) setOffset((pageIndex + 1) * pageSize);
+  };
+
+  const handlePrevious = () => {
+    previousPage();
+    if (setOffset) setOffset((pageIndex - 1 || 0) * pageSize);
+  };
+
+  useEffect(() => {
+    if (setSortBy) setSortBy(sortBy);
+  }, [sortBy, setSortBy]);
+
+  return (
+    <>
+      <ChakraTable {...getTableProps()}>
+        <Thead>
+          <Tr>
+            {allColumns.map((column) => (
+              <Th
+                {...column.getHeaderProps(column.getSortByToggleProps())}
+              >
+                {column.render('Header')}
+                <chakra.span pl="2">
+                  {column.isSorted && (
+                    column.isSortedDesc ? (
+                      <MdArrowDropDown aria-label="sorted descending" style={{ display: 'inline' }} size="2em" />
+                    ) : (
+                      <MdArrowDropUp aria-label="sorted ascending" style={{ display: 'inline' }} size="2em" />
+                    )
+                  )}
+                </chakra.span>
+              </Th>
+
+            ))}
+          </Tr>
+        </Thead>
+        <Tbody {...getTableBodyProps()}>
+          {!data.length && (
+          <Tr>
+            <Td colSpan={2}>No Data found.</Td>
+          </Tr>
+          )}
+          {page.map((row) => {
+            prepareRow(row);
+            return (
+              <Tr
+                {...row.getRowProps()}
+                _odd={{ backgroundColor: oddColor }}
+                _hover={{ backgroundColor: hoverColor }}
+              >
+                {row.cells.map((cell) => (
+                  <Td
+                    {...cell.getCellProps()}
+                    py={3}
+                  >
+                    {cell.render('Cell')}
+                  </Td>
+                ))}
+              </Tr>
+            );
+          })}
+        </Tbody>
+      </ChakraTable>
+      <Flex alignItems="center" justifyContent="flex-end">
```

Comment:
The pagination gets kind of lost aligned to the right. I'd suggest aligning to the left and adding some margin.
```suggestion
      <Flex alignItems="center" justifyContent="flex-start" my={4}>
```